app/controlplane/pkg/biz/organization.go

```diff
@@ -244,3 +248,79 @@ func (uc *OrganizationUseCase) Delete(ctx context.Context, id string) error {
 	// Delete the organization
 	return uc.orgRepo.Delete(ctx, orgUUID)
 }
+
+// AutoOnboardOrganizations creates the organizations specified in the onboarding config and assigns the user to them
+// with the specified role if they are not already a member.
+func (uc *OrganizationUseCase) AutoOnboardOrganizations(ctx context.Context, userID string) error {
+	// Parse user UUID
+	userUUID, err := uuid.Parse(userID)
+	if err != nil {
+		return NewErrInvalidUUID(err)
+	}
+
+	for _, spec := range uc.onboardingConfig {
+		// Ensure the organization exists or create it if it doesn't
+		org, err := uc.ensureOrganizationExists(ctx, spec)
+		if err != nil {
+			return fmt.Errorf("failed to ensure organization exists: %w", err)
+		}
+
+		// Parse organization UUID
+		orgUUID, err := uuid.Parse(org.ID)
+		if err != nil {
+			return NewErrInvalidUUID(err)
+		}
+
+		// Ensure user membership
+		if err := uc.ensureUserMembership(ctx, orgUUID, userUUID, spec); err != nil {
+			return fmt.Errorf("failed to ensure user membership: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// ensureOrganizationExists ensures that an organization specified by the onboarding configuration exists.
+// If the organization does not exist, it creates it.
+func (uc *OrganizationUseCase) ensureOrganizationExists(ctx context.Context, spec *conf.OnboardingSpec) (*Organization, error) {
+	// Ensure the organization exists or create it if it doesn't
+	org, err := uc.orgRepo.FindByName(ctx, spec.GetName())
+	if err != nil {
+		return nil, fmt.Errorf("failed to find organization: %w", err)
+	} else if org == nil {
```

Comment:
I might be wrong, but I remember we talked offline about about not creating orgs, and assume they existed previously.

Reply:
As far as I remember if it was because of the possible collisions it was solved on the DB. In any case we can always come back and remove it.

Maybe when we have https://github.com/chainloop-dev/chainloop/issues/945 done.

Reply:
True, I'd do it in another smaller follow up PR